journal/src/View/Converter/Block/ParagraphConverter.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Libero\Journal\View\Converter\Block;
+
+use Libero\Journal\Dom\Element;
+use Libero\Journal\View\Converter\ViewConverter;
+use const Libero\Journal\LIBERO;
+
+final class ParagraphConverter implements ViewConverter
+{
+    private $converter;
+
+    public function __construct(ViewConverter $converter)
+    {
+        $this->converter = $converter;
+    }
+
+    /**
+     * @param Element $object
+     */
+    public function convert($object, array $context = []) : string
+    {
+        $context['lang'] = $context['lang'] ?? null;
+
+        $attributes = '';
+        if ($object->getAttribute('lang') ?? $context['lang'] !== $context['lang']) {
+            $attributes .= " lang=\"{$object->getAttribute('lang')->toText()}\"";
+            $context['lang'] = $object->getAttribute('lang')->toText();
+        }
+
+        $text = '';
+        foreach ($object as $child) {
+            $text .= $this->converter->convert($child, $context);
+        }
+
+        return <<<EOT
```

Comment:
the `ViewConverter` instances do not use patterns; is it a temporary simplification or an active decision?

Reply:
Patterns weren't part of the walking skeleton, just plain HTML.